cdap-ui/app/cdap/components/PipelineDetails/RunLevelInfo/RunStatus.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import StatusMapper from 'services/StatusMapper';
+import IconSVG from 'components/IconSVG';
+import RunningRunsPopover from 'components/PipelineDetails/RunLevelInfo/RunningRunsPopover';
+
+const mapStateToProps = (state) => {
+  return {
+    runs: state.runs,
+    currentRun: state.currentRun,
+    pipelineId: state.name
+  };
+};
+
+const RunStatus = ({runs, currentRun, pipelineId}) => {
+  let status;
+  if (currentRun && currentRun.status) {
+    status = currentRun.status;
+  } else {
+    status = 'DEPLOYED';
+  }
+  status = StatusMapper.lookupDisplayStatus(status);
+  let statusCSSClass = StatusMapper.getStatusIndicatorClass(status);
+
+  let runningRuns = runs.filter(run => run.status === 'RUNNING');
```

Comment:
I see the run states being used as string. Would be better if we externalize it and use it as an enum (or constant). Easier to change if backend changes at any point.

Reply:
Hmm if we change this then it makes sense to change `StatusMapper` as well. Do you want me to do that?